extensions/functions_arithmetic.yaml

```diff
@@ -441,6 +441,57 @@ aggregate_functions:
         decomposable: MANY
         intermediate: fp64?
         return: fp64?
+  - name: "product"
+    description: Product of a set of values.
+    impls:
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: i8
+        nullability: MIRROR
+        decomposable: MANY
+        intermediate: i64?
+        return: i8?
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: i16
+        nullability: MIRROR
+        decomposable: MANY
+        intermediate: i64?
+        return: i16?
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: i32
+        nullability: MIRROR
+        decomposable: MANY
+        intermediate: i64?
+        return: i32?
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: i64
+        nullability: MIRROR
+        decomposable: MANY
+        intermediate: i64?
+        return: i64?
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: fp32
+        nullability: MIRROR
+        decomposable: MANY
+        intermediate: fp64?
+        return: fp32?
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: fp64
+        nullability: MIRROR
+        decomposable: MANY
+        intermediate: fp64?
+        return: fp64?
```

Comment:
I missed this before, but these should also have the floating-point rounding options rather than overflow.

Reply:
Included floating-point rounding options